hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/WriteStatus.java

```diff
@@ -87,12 +90,12 @@ public WriteStatus() {
    * Mark write as success, optionally using given parameters for the purpose of calculating some aggregate metrics.
    * This method is not meant to cache passed arguments, since WriteStatus objects are collected in Spark Driver.
    *
-   * @param record deflated {@code HoodieRecord} containing information that uniquely identifies it.
+   * @param recordDelegateLazy     HoodieKey with location information.
    * @param optionalRecordMetadata optional metadata related to data contained in {@link HoodieRecord} before deflation.
    */
-  public void markSuccess(HoodieRecord record, Option<Map<String, String>> optionalRecordMetadata) {
+  public void markSuccess(Lazy<HoodieRecordDelegate> recordDelegateLazy, Option<Map<String, String>> optionalRecordMetadata) {
```

Comment:
If we keep the instantTime attribute inside `WriteStatus`, the api can be simplified as just passing around a `HoodieKey`, because the partitionPath and fileId are already deterministic for a `WriteStatus`.

Reply:
we might have to embed isDelete info as well. So, not sure if we can get rid of HoodieRecordDelegate. just that we did not fix it in this patch.

Reply:
we still need some extensibility here, as Siva's comment suggested. There is a follow up fix for RLI delete which would require that.